src/langcheck/metrics/ja/reference_free_text_quality.py

```diff
@@ -116,40 +107,19 @@ def _sentiment_local(generated_outputs: List[str]) -> List[float]:
 
     Args:
         generated_outputs: A list of model generated outputs to evaluate
+        overflow_strategy: The strategy to handle inputs that are longer than
+            the maximum input length of the model.
 
     Returns:
         A list of scores
     '''
-    global _sentiment_tokenizer, _sentiment_model
-
-    if _sentiment_tokenizer is None or _sentiment_model is None:
-        _sentiment_tokenizer = AutoTokenizer.from_pretrained(
-            _sentiment_model_path)
-
-        # There is a "Some weights are not used warning" but we ignore it
-        # because that is intended.
-        with _handle_logging_level():
-            _sentiment_model = (AutoModelForSequenceClassification.
-                                from_pretrained(_sentiment_model_path))
-
-    input_tokens = _sentiment_tokenizer(generated_outputs,
-                                        return_tensors='pt',
-                                        padding=True)
-
-    batch_size = 8
-    scores = []
-    with torch.no_grad():
-        for i in tqdm_wrapper(range(0, len(generated_outputs), batch_size),
-                              total=(len(generated_outputs) + batch_size - 1) //
-                              batch_size):
-            batch_input_tokens = {
-                k: v[i:i + batch_size] for k, v in input_tokens.items()
-            }
-            # Probabilities of [negative, neutral, positive]
-            probs = torch.nn.functional.softmax(
-                _sentiment_model(**batch_input_tokens).logits, dim=1)
-            scores.extend((probs[:, 1] / 2 + probs[:, 2]).tolist())
-    return scores
+    scorer = AutoModelForSequenceClassificationScorer(
+        language='ja',
+        metric='sentiment',
+        class_weights=[0, 0.5, 1],
+        overflow_strategy=overflow_strategy,
+        max_input_length=512)
```

Comment:
(Just curious) why do we sometimes need to specify this manually? Does this not have a `config.max_position_embeddings`?

Reply:
Seemingly some models are not accurate about their `config.max_position_embeddings`. For example, this model has `max_position_embeddings=514` but it doesn't seem to accept input tokens longer than 514.
https://huggingface.co/cardiffnlp/twitter-xlm-roberta-base-sentiment-multilingual/blob/main/config.json

There are other models that do not define `max_input_length` too, that's why I made it one of the init arguments.